File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product
def solution(word):
    alphabet = ['A', 'E', 'I', 'O', 'U']
    arr = []
    for i in range(1,6):
        for p in product(alphabet, repeat = i):
            arr.append(''.join(p))
    arr.sort()
    return arr.index(word)+1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    const words = ['A','E','I','O','U']
    let dict = [...words]
    for (let i = 1; i<5; i++){
        for (const d of dict){
            if (d.length === i){
                for (const w of words){
                    dict.push(d+w)
                }
            }
        }
    }
    dict.sort();
    return dict.indexOf(word)+1;
}